plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1278,4 +1292,57 @@ public void testDeleteWithVarcharPredicate()
     {
         throw new SkipException("This is implemented by testDeleteWithVarcharEqualityPredicate");
     }
+
+    @Test(dataProvider = "testInsertBatchSizeSessionProperty")
+    public void testInsertBatchSizeSessionProperty(Integer batchSize, Integer numberOfRows)
+    {
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), "insert_batch_size", batchSize.toString())
+                .build();
+
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_insert_batch_size",
+                "(a varchar(36), b bigint)")) {
+
+            String values = String.join(",", makeValuesForInsertBatchSizeSessionPropertyTest(numberOfRows));
+            assertUpdate(session, "INSERT INTO " + table.getName() + " (a, b) VALUES " + values, numberOfRows);
+            assertQuery("SELECT COUNT(*) FROM " + table.getName(), format("VALUES %d", numberOfRows));
+        }
+    }
+
+    private static List<String> makeValuesForInsertBatchSizeSessionPropertyTest(int numberOfRows) {
+        List<String> result = new ArrayList<>(numberOfRows);
+        for (int i = 0; i < numberOfRows; i++) {
+            result.add(String.format("('%s', %d)", UUID.randomUUID(), ThreadLocalRandom.current().nextLong()));
```

Comment:
nit: Always static import `String.format`.

Reply:
Fixed, forgot to fire 'mvn test' one last time before push.